In [77]:
import pandas as pd
import numpy as np

In [86]:
old = pd.read_csv('blink_results_old.csv')
new = pd.read_csv('blink_results_new.csv')

In [87]:
new_ids, new_counts = np.unique(new['doc_idx'], return_counts=True)
count_new = {new_ids[i]: new_counts[i] for i in range(len(new_ids))}
old_ids, old_counts = np.unique(old['doc_idx'], return_counts=True)
count_old = {old_ids[i]: old_counts[i] for i in range(len(old_ids))}

In [88]:
sample = pd.read_csv('../../sampling/FAA_sample_100.csv')

In [89]:
overlap = {}
for doc_idx in sample['Unnamed: 0']:
    if doc_idx in count_new and doc_idx not in count_old:
        print(f"New id {doc_idx}")
    elif doc_idx in count_old and doc_idx not in count_new:
        print(f"Throw out {doc_idx}")
    elif doc_idx in count_old and doc_idx in count_new:
        overlap[doc_idx] = {'old':old[old['doc_idx'] == doc_idx].index, 'new':new[new['doc_idx'] == doc_idx].index}

Throw out 2600
Throw out 1612


In [90]:
for doc_idx in overlap:
    if not list(old.loc[overlap[doc_idx]['old']]['mention']) == list(new.loc[overlap[doc_idx]['new']]['mention']):
        print(f"Discrepancy in {doc_idx}")

Discrepancy in 2685
Discrepancy in 2334


In [91]:
doc_idx = 2685
old.loc[overlap[doc_idx]['old']]['mention'], new.loc[overlap[doc_idx]['new']]['mention']

(1097    ray airport
 1098           pilo
 Name: mention, dtype: object,
 1050    ray airport
 Name: mention, dtype: object)

In [76]:
doc_idx = 2334
old.loc[overlap[doc_idx]['old']]['mention'], new.loc[overlap[doc_idx]['new']]['mention']

(764    eastern standard
 765              skyway
 Name: mention, dtype: object,
 738    eastern standard
 739      skyway airline
 Name: mention, dtype: object)

### Summary

Changes between old and new versions include:
- new version does not contain 2600 and 1612
- On 2685, pilo is not longer recognized as an entity
- On 2334, skyway is now skyway airline